fs/ftpfs.py

```diff
@@ -728,9 +728,8 @@ def _scandir(self, path, namespaces=None):
                     for raw_info in self._parse_mlsx(lines):
                         yield Info(raw_info)
                     return
-            with self._lock:
-                for info in self._read_dir(_path).values():
-                    yield info
+            for info in self._read_dir(_path).values():
+                yield info
```

Comment:
Does this change have any impact on non-pypy usages of pyfilesystem?

Reply:
I think this was purely because the lock is redundant, as there is a lock on L714.